components/ingest-service/migration/migration.go

```diff
@@ -256,11 +257,28 @@ func (ms *Status) migrateNodeStateToCurrent(previousIndex string) error {
 	ms.taskCompleted()
 
 	ms.update(fmt.Sprintf("Reindexing %s index to current", previousIndex))
-	err := ms.client.ReindexNodeStateToLatest(ms.ctx, previousIndex)
+	reindexTaskID, err := ms.client.ReindexNodeStateToLatest(ms.ctx, previousIndex)
 	if err != nil {
 		ms.updateErr(err.Error(), "Unable to reindex node-state to latest")
 		return err
 	}
+
+	// Wait for Reindex task to complete
+	for {
```

Comment:
Should we time out of this loop? We don't really know how long this could take. If elasticsearch restarts then this loop would fail at line 273.

Reply:
what would the benefit of timing out give us?  is there any chance this would just go on forever and ever?

Reply:
I don't think there would any benefit. This would not go on forever because the job has to end in elasticsearch at some point.